core/src/main/scala/org/http4s/util/Renderable.scala

```diff
@@ -143,3 +146,18 @@ class StringWriter(size: Int = 64) extends Writer {
 
   def result(): String = sb.toString
 }
+/** [[Writer]] that will result in a `ByteVector`
+  * @param bv initial ByteVector`
+  */
+case class ByteVectorWriter(private var bv: ByteVector = ByteVector.empty,
```

Comment:
n.b. The var nonsense is required by the `this.type`.  Otherwise, we'd just return copies.